Add support for multi author blog posts

notmandatory opened the pull request:
I've been wanting to clean up multi author support and @RCasatta's issue #34 convinced me to try to do it.

I followed these instructions: https://www.netlify.com/blog/2018/07/24/hugo-tips-how-to-create-author-pages/

New author pages have additional links and optional bios and pictures if anyone wants to add them (though it's untested).

notmandatory committed:
Add support for multi author blog posts

afilini commented:
Everything looks really good, the only nit I have is that there's a whitespace before the comma in the list of authors. I don't really know Hugo all that well, but replacing some `{{ }}` with `{{- -}}` seems to fix it. This should be enough:

```diff
diff --git a/layouts/blog/list.html b/layouts/blog/list.html
index 6f53037..128b9e9 100644
--- a/layouts/blog/list.html
+++ b/layouts/blog/list.html
@@ -14,12 +14,12 @@
             {{ if isset .Params "authors" }}
                 <span>By </span>
                 {{- range $index, $author := .Params.authors }}
-                  {{- with $.Site.GetPage "taxonomyTerm" (printf "authors/%s" (urlize .)) }}
-                    {{ if gt $index 0 }}
+                  {{- with $.Site.GetPage "taxonomyTerm" (printf "authors/%s" (urlize .)) -}}
+                    {{- if gt $index 0 -}}
                       <span>, </span>
-                    {{ end }}
+                    {{- end -}}
                     <span><a href="{{ "/authors/" | relLangURL }}{{ .Params.name | urlize }}">{{ .Params.name }}</a></span>
-                  {{ end }}
+                  {{- end -}}
                 {{ end }}
             {{ end }}
 
diff --git a/layouts/partials/header.html b/layouts/partials/header.html
index 427face..d4d3391 100644
--- a/layouts/partials/header.html
+++ b/layouts/partials/header.html
@@ -102,12 +102,12 @@
             {{ if isset .Params "authors" }}
                 <span class="blog-author">By </span>
                 {{- range $index, $author := .Params.authors }}
-                  {{- with $.Site.GetPage "taxonomyTerm" (printf "authors/%s" (urlize .)) }}
-                    {{ if gt $index 0 }}
+                  {{- with $.Site.GetPage "taxonomyTerm" (printf "authors/%s" (urlize .)) -}}
+                    {{- if gt $index 0 -}}
                       <span class="blog-author">, </span>
-                    {{ end }}
+                    {{- end -}}
                     <span class="blog-author"><a href="{{ "/authors/" | relLangURL }}{{ .Params.name | urlize }}">{{ .Params.name }}</a></span>
-                  {{ end }}
+                  {{- end -}}
                 {{ end }}
             {{ end }}
             {{ else }}

```

notmandatory committed:
Remove space before comma in authors list

notmandatory committed:
Add /authors page

notmandatory commented:
> Everything looks really good, the only nit I have is that there's a whitespace before the comma in the list of authors. I don't really know Hugo all that well, but replacing some `{{ }}` with `{{- -}}` seems to fix it. This should be enough:
> 

Good eye! thanks I added your fix. I also added a `/authors` page, not linked anywhere but added for completeness.
